src/dynamic_boundary_conditions/hirds_depth_data_to_db.py

```diff
@@ -5,113 +5,144 @@
-    return hirds_data
-
-
-def add_hirds_depth_data_to_db(path: str, site_id: str, engine):
-    """Store each site's depth data in the database. Each csv file contains
-    data for historical rainfall depth, and for various rcp and time period.
-    To view the csv file, go to : https://hirds.niwa.co.nz/, select a site and
-    generate a report, there are rainfall depths for diffrent RCP Scenarios.
+    return insp.has_table(db_table_name, schema="public")
+
+
+def get_sites_id_in_catchment(engine, catchment_polygon: Polygon) -> list:
+    """Get rainfall sites id within the catchment area from the 'rainfall_sites' table in the database."""
+    query = f"""SELECT * FROM rainfall_sites AS rs
+        WHERE ST_Intersects(rs.geometry, ST_GeomFromText('{catchment_polygon}', 4326))"""
+    sites_in_catchment = gpd.GeoDataFrame.from_postgis(query, engine, geom_col="geometry", crs=4326)
+    sites_id_in_catchment = sites_in_catchment["site_id"].tolist()
+    return sites_id_in_catchment
+
+
+def get_sites_id_not_in_db(engine, sites_id_in_catchment: list) -> list:
+    """Get the list of sites that are in the catchment area but are not available in the database."""
+    query = "SELECT DISTINCT site_id FROM rainfall_depth;"
+    # Get dataframe of sites in the database
+    sites_id_in_db = pd.read_sql_query(query, engine)
+    # Convert dataframe to list of sites in the database
+    sites_id_in_db = sites_id_in_db["site_id"].tolist()
+    # yields the elements in `sites_id_in_catchment` that are NOT in `sites_id_in_db`
+    sites_id_not_in_db = list(set(sites_id_in_catchment).difference(sites_id_in_db))
+    return sites_id_not_in_db
+
+
+def get_layout_structure_of_csv(filepath) -> list:
+    """Read the csv file of each site's rainfall data and return a list of tuples (skip_rows, rcp, time_period)
+    of its layout structure"""
+    skip_rows = []
+    rcp = []
+    time_period = []
+    # Read file line by line with a for loop
+    with open(filepath) as file:
+        for index, line in enumerate(file):
+            # Get lines that contain "(mm) ::" for depth data or "(mm/hr) ::" for intensity data
+            if (line.find("(mm) ::") != -1) or (line.find("(mm/hr) ::") != -1):
+                # Add the row number to skip_rows list
+                skip_rows.append(index + 1)
+                # Add the obtained rcp and time_period values to list
+                rcp_result = re.search(r"(\d*\.\d*)", line)
+                period_result = re.search(r"(\d{4}-\d{4})", line)
+                if rcp_result or period_result is not None:
+                    rcp.append(float(rcp_result[0]))
+                    time_period.append(period_result[0])
+                else:
+                    rcp.append(float('nan'))
+                    time_period.append(None)
+    # Merge the three different lists into one list of tuples
+    layout_structure = list(zip(skip_rows, rcp, time_period))
+    return layout_structure
```

Comment:
Note for self: Luke to come back to this after typing annotation is complete.

It is a good use of RE I think, I am going to come back about the use of list and tuples later